[Feature Cleanup] APIServerIdentity to Node Conformance

mariafromano-25 opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it: 
The APIServerIdentity feature is Beta since v1.26 and is enabled by default on clusters. Verified by creating a new kind cluster with default node image v1.34.0 and ran: 

` $~ kubectl get --raw /metrics | grep APIServerIdentity  kubernetes_feature_enabled{name="APIServerIdentity",stage="BETA"} 1`

#### Which issue(s) this PR is related to:
Partial fix for https://github.com/kubernetes/kubernetes/issues/134172

#### Special notes for your reviewer:
FeatureGates is a separate cleanup
corresponding changes will be made in k/test-infra

#### Does this PR introduce a user-facing change?
no

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```
/assign @SergeyKanzhelev @roycaihw

mariafromano-25 committed:
cleaning up apiserveridentity featuer to node conformance since it's beta

SergeyKanzhelev commented on test/e2e/apimachinery/apiserver_identity.go:
```diff
@@ -80,7 +79,7 @@ func restartAPIServer(ctx context.Context, node *v1.Node) error {
 
 // This test requires that --feature-gates=APIServerIdentity=true be set on the apiserver
-var _ = SIGDescribe("kube-apiserver identity", feature.APIServerIdentity, func() {
+var _ = SIGDescribe("kube-apiserver identity", framework.WithNodeConformance(), func() {
```
yes, in this case it is since it is not a Node test

mariafromano-25 committed:
remove Node Conformance since this test is not for node, but still remove feature label for cleanup

Removed 'WithNodeConformance' from SIGDescribe for kube-apiserver identity test.

mariafromano-25 committed:
removed apiserveridentity from feature file since only used once and removed
